deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/TwinTags.java

```diff
@@ -35,7 +35,7 @@ protected TwinTags()
     protected synchronized JsonElement update(Map<String, Object> tagsMap) throws IllegalArgumentException
     {
         JsonElement innerDiff = updateFromMap(tagsMap, tags);
-        if((innerDiff == null) || (innerDiff.toString().equals("{}")))
```

Comment:
updateFromMap never returns a null JsonElement (it returns a new or modified JsonObject)